probe/ebpf/flow.c

```diff
@@ -342,15 +414,15 @@ static inline void fill_link(struct __sk_buff *skb, int offset, struct flow *flo
 	update_hash_half(&hash_dst, layer->mac_dst[2] << 8 | layer->mac_dst[3]);
 	update_hash_half(&hash_dst, layer->mac_dst[4] << 8 | layer->mac_dst[5]);
 
+	layer->id = load_half(skb, offsetof(struct ethhdr, h_proto));
```

Comment:
seems to be incorrect, should be filled with the vlan id.

Reply:
Also I'm seeing flows with only ethernet(no ARP, just broadcast) layer which seems to be incorrect.

Reply:
link->id is filled by ethernet layer and vlan layers

Reply:
Seeing wrong link->id as well on some flows